src/components/Calculator/index.jsx

```diff
@@ -6,22 +6,47 @@ import Result from '../Result';
 
 const Calculator = () => {
     const[currentNumber, setCurrentNumber] = React.useState('');
-    const[currentOperation, setCurrentOperation] = React.useState([]);
+    const[currentOperations, setCurrentOperation] = React.useState([]);
+    const[saveOperation, setSaveOperation] = React.useState(false);
 
-    let handlerCurrentOperation = (elementOperation) => {
-        setCurrentOperation([...currentOperation, elementOperation]);
+    let handlerCurrentOperation = (operand, operator) => {
+        setCurrentOperation([...currentOperations, operand, operator]);
+    }
+
+    let handlerNumberReset = () => {
+        setCurrentNumber('');
     }
 
     let handlerCurrentNumber = (number) => {
+        setSaveOperation(true);
         setCurrentNumber(currentNumber + number);
     }
 
     let handlerResult = () => {
         setCurrentNumber('');
     }
 
+    let showOperation = () => {
+        let operation = '';
+
+        currentOperations.map((value) => (
+            operation+=value
+        ));
+
+        return operation;
+    }
+
+    let saveOperations = (operand, operator) => {
+        if (saveOperation) {
+            handlerCurrentOperation(operand, operator);
+            handlerNumberReset();
+            setSaveOperation(false);
+        }
+    }
+
```

Comment:
I have a doubt here, I need to do more than one action when a Operator is clicked, and to don't send many functions to the onClick() I made a function with that instructions, including the function to set the state.

I send to the component to exec in onClick() in the button, can you tell me if that is a correct form? 
If not, how would it be? 

@baquiax

Reply:
This is ok, it's a usual approach :) The callback can  run any amount of instructions.